14_contains/solution/contains-solution.js

```diff
@@ -1,19 +1,22 @@
-const contains = function (obj, searchValue) {
-  const values = Object.values(obj);
+const contains = function (initialObject, searchValue, queue = []) {
+  if (initialObject !== null) queue.push(initialObject);
+  if (initialObject === null && queue.length === 0) return false;
+  const item = queue.shift();
+  const values = Object.values(item);
 
   // NaN === NaN evaluates to false
   // Normally, we would have to do an explicit Number.isNaN() check to compare NaN equality
   // However, Array.prototype.includes automatically handles this for us
   if (values.includes(searchValue)) return true;
 
   const nestedObjects = values.filter(
+    // typeof null === 'object' evaluates to true ¯\_(ツ)_/¯
     (value) => typeof value === "object" && value !== null,
   );
-  for (const nestedObject of nestedObjects) {
-    return contains(nestedObject, searchValue);
-  }
 
-  return false;
+  const newQueue = queue.concat(nestedObjects);
+
+  return contains(null, searchValue, newQueue);
```

Comment:
I think `some()` is a lot simpler than using a queue and should pass all the specs. Also, "queue" is not something learners are aware about while doing these exercises. It's not introduced until the Common Data Structures and Algorithms lesson, which comes after these exercises.

```js
const contains = function(object, searchValue) {
  const values = Object.values(object);
  
  if (values.includes(searchValue)) return true;
  
  const nestedObjects = values.filter((value) => typeof value === "object" && value !== null);
  return nestedObjects.some((nestedObject) => contains(nestedObject, searchValue));
}
```

Reply:
Oh yeah, thanks - you solution is certainly a lot cleaner. I have never considered the `some` method but turns out that it is actually really useful.